### DISCUSSION TITLE
Appropriateness of questions

### DISCUSSION THREAD

**Discussion by lafinsd** [HexFiend/HexFiend#379] [Q&A]
I am not C++ fluent at all. I'm a retired firmware developer who worked on deeply embedded control systems exclusively in 'C'. I'm not familiar with the C++ environment or standards and best practices. GitHub either, for that matter.

I sometimes have questions about how Hex Fiend works that could probably be answered by reading the code. Is it appropriate for me to ask such questions here?

Thanks.

**Comment by kainjow** [comment]
Sure that is totally fine. I just make no guarantees on timeliness of responses 🙂

**Reply by lafinsd** [reply]
Thanks...

Running Hex Fiend from the .../MacOS directory using Terminal lets me read stdin in a Binary Template. If I run Hex Fiend as a regular application and invoke the same Template the Template appears to ignore the input request and otherwise execute correctly. Somehow the read of stdin is transparent. When running HF as a regular application what happens to the value of the variable that is now not read from stdin?
